src/DGtal/geometry/doc/moduleDigitalConvexity.dox

```diff
@@ -0,0 +1,165 @@
+
+A last useful notion is the \e subconvexity. Let \f$ X \subset
+\mathbb{Z}^d \f$ some arbitrary digital set. Then the digital set \f$
+S \subset \mathbb{Z}^d \f$ is said to be \b digitally \a k- \b
+subconvex to \f$ X \f$ whenever \f$ C^d_k \lbrack \mathrm{Conv}(S)
+\rbrack \subset C^d_k \lbrack X \rbrack \f$. And \f$ S \f$ is said to
+be \b fully (digitally) \b subconvex to \f$ X \f$ whenever it is
+digitally \a k- subconvex to \f$ X \f$ for \f$ 0 \le k \le d \f$.
+
+Subconvexity is a useful for notion for digital contour and surface
+analysis. It tells which subsets of these digital sets are \e tangent
+to them.
+
+@section dgtal_dconvexity_sec2  Checking digital convexity
+
+Three classes help to check digital convexity.
+
+- BoundedLatticePolytope is the class that is used to build polytopes,
+  i.e. intersections of half-spaces, which are a way to represent
+  convex polyhedra.
+
+- CellGeometry is used to store sets of cells and provides methods to
+  build the set of cells that intersect a polytope or the set of cells
+  that touches a set of digital points.
+
+- DigitalConvexity provides many helper methods to build
+  BoundedLatticePolytope s and CellGeometry s and to check digital
+  convexity and subconvexity.
+
+
+@subsection dgtal_dconvexity_sec21 Building a polytope
+
+For now, the class is quite limited. You may build a polytope from a
+range of \f$ d + 1 \f$ points in general position. The polytope is
+then a simplex. Otherwise you may provide a range of half-spaces. 
+
+You may also cut a polytope by a new halfspace
+(BoundedLatticePolytope::cut), count the number of lattice points
+inside, interior or on the boundary (BoundedLatticePolytope::count,
+BoundedLatticePolytope::countInterior,
+BoundedLatticePolytope::countBoundary) or enumerate them.
+
+\code
+using namespace DGtal::Z3i;
+typedef BoundedLatticePolytope< Space > Simplex;
+Simplex S( { Point(0,0,0), Point(3,0,0), Point(1,5,0), Point(-3,2,4) } );
+std::cout << S.count() << std::endl;
+simplex += Simplex::UnitSegment( 0 ); // Extend it along x
+std::cout << S.count() << std::endl;
+\endcode 
+
+Last you may compute Minkowski sums of a polytope with axis-aligned
+segments, squares or (hyper)-cubes (BoundedLatticePolytope::operator+=). 
+
+@Note You can check if the result of a Minkowski sum will be valid by
+calling BoundedLatticePolytope::canBeSummed before. The support is for
+now limited to polytopes built as simplices in 2D and 3D.
+
+@see DigitalConvexity::makeSimplex
+
+@subsection dgtal_dconvexity_sec22 Building a set of cells from digital points
+
+The class CellGeometry can compute and store set of cells of different
+dimensions. You specify at construction a Khalimsky space (any model
+of concepts::CCellularGridSpaceND) as well as the dimensions of the
+cells you are interested in.
+
```

Comment:
Maybe we should clarify this class wrt to the cells container available in the KhalimskySpace classes.

Reply:
This one embeds to topological structure to R^n but maybe we should clarify

Reply:
Add comment that it uses a CubicalComplex as internal representation.